### PR TITLE
Allow touches on overflowed children when `overflow: visible` (Android)

### PR DESCRIPTION
**Pull request by danielgindi**
## Summary

Currently `overflow: visible` allows showing child views that are outside the parent's bounds (well, `overflow: visible` on the grandparent allows that, not the parent).
But those non-clipped children do not receive touches.

This is an attempt to solve it.

## Changelog

[Android] [Fixed] - `overflow: visible` now allows interaction with children that are rendered outside their parents' bounds.

## Test Plan

* Added `overflow: visible`, tapped on a button.
* Added a `MapView`, tested scrolling the map while overflowed elements are visible over it, and tapping on those elements.

### CODE REVIEW

**Comment by danielgindi** [comment]
For some reason the CI fails, without any details.

**Comment by nick-michael** [comment]
Can anyone trigger a rebuild of this? Maybe was just a bad day for CI?

**Comment by danielgindi** [comment]
@nick-michael Rebasing on latest stable

**Comment by danielgindi** [comment]
I'm guessing another bad day for CI?

**Comment by nick-michael** [comment]
Haha guess not, I don't want this to get forgotten about though, this is such a welcome change for RN android

**Comment by danielgindi** [comment]
It's amusing that the iOS tests are the ones that failing. The changes are Android only.
Due to the fact that this **** is impossible to compile anyway, I had to move the classes to a separate project, work on them there, and apply the diff to the real RN project. Now I can only hope that this didn't break when rebasing on `master` again and again over time.

**Comment by ltcaosj** [comment]
We really need to PR merged :). Same stuff is working fine on iOS, not on Android.

**Comment by kdenz** [comment]
Yes we'll need this urgently too, because we have an expandable button which will go overflow and open an animated view action menu, and right now on android users cannot interact with the expanded content directly. Changing `pointerEvents` and `zIndex` doesn't work.

**Comment by ltcaosj** [comment]
@danielgindi What should we do to resolve the CI issue? Your fixing is so much import to us and the react-native community. After fixing this one we can create a complex UI much easier.

**Comment by danielgindi** [comment]
@ltcaosj I have no idea. RN does not compile easily, and I can't spend too much time forcing it to and then troubleshooting their CI scripts. I did not even touch the iOS code, but the CI is failing on iOS. So I'm not going to even try to solve this, as it's not my issue.

**Comment by fabriziobertoglio1987** [comment]
***ci/circleci: js_coverage — Your tests failed on CircleCI***

```
#!/bin/bash -eo pipefail
yarn test --coverage --maxWorkers=2
cat ./coverage/lcov.info | ./node_modules/.bin/coveralls
```
All test succed,

```
Test Suites: 102 passed, 102 total
Tests:       1 skipped, 981 passed, 982 total
Snapshots:   477 passed, 477 total
Time:        94.842s
Ran all test suites.
Done in 95.74s.
```
but command `/home/circleci/react-native/node_modules/coveralls/bin/coveralls.js` fails.
then returns 

``
/home/circleci/react-native/node_modules/coveralls/bin/coveralls.js:18
        throw err;
        ^
Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}
Exited with code 1
```

this errors is connected to misconfiguration inside circleci of the `coveralls` js library.

**Comment by tarasvakulka** [comment]
Hi, may be somebody know when this PR will be merged ?

**Comment by danielgindi** [comment]
@kelset @alloy I know you guys are busy, but this solves a fundamental issue with Android... Can you please take the time to review?

**Comment by kelset** [comment]
Hey Daniel, thanks for the ping. I'll try to put this in front of the right people but can't make promises 😓

**Comment by EzeMortal** [comment]
I need this commit :'(

**Comment by TheMrZZ** [comment]
Actually, this would help us a lot. As Daniel said, this solves a fundamental problem in Android. If anyone could take a look and see what's wrong, it would help a lot.

**Comment by Owpac** [comment]
I really need this bug to be fixed! Can someone try to push this code again (with the latest react-native version)? Maybe the CI problem was resolved since then.

**Comment by danielgindi** [comment]
@kelset I've rebased on the latest point in `master` where the CI seems to work. Please look at it, I'm starting to get really spammed by users approaching me on email and by facebook messenger asking me how to fix this issue...

**Comment by aanation** [comment]
Any news?
So many people need this fix :(
It is very difficult for me to develop a custom autocomplete, a dropdown and many other components where absolute positioning is required :(

**Comment by 1awaleed** [comment]
I have made a workaround for my custom dropdowns but involves to much extra logic & very limiting in reusability of components. I really need this.

**Comment by ltcaosj** [comment]
@1awaleed yeah, we all need this capability :(

**Comment by peetklecha** [comment]
i just want to chime in to say that i too am waiting on this functionality -- would love for this to be approved!

**Comment by daxfrost** [comment]
I'm not sure if it's the same thing, but I don't seem to get touch events on overflowed sub views on iOS either... am i barking up the wrong tree?

**Comment by danielgindi** [comment]
> I'm not sure if it's the same thing, but I don't seem to get touch events on overflowed sub views on iOS either... am i barking up the wrong tree?
> 
> 
> 
> image
> 
> 
> 
> The bottom portion of a view / button / touchable view, like above, is not tappable.

In ios there is no issue with this. If you can’t seem to touch it, it is either covered by another view stealing the touches, or scrolled under a “not safe” area, where status bar or something else steals the touches.

**Comment by Rodrigo816** [comment]
I really need this. When its solved? any news? what workarounds can I do?

**Comment by Ragnar-H** [comment]
@Rodrigo816 I guess you could [build this fix from source](https://github.com/facebook/react-native/wiki/Building-from-source) as a workaround.

Then you can also verify if this PR fixes the issue 🤓

**Comment by qalqi** [comment]
It's a Bird, It's a Plane, It's clipChildren==false

**Comment by shevchenkona19** [comment]
I just bumped into this issue. It's so important. Waiting on the merge too!

**Comment by danielgindi** [comment]
> @Rodrigo816 I guess you could [build this fix from source](https://github.com/facebook/react-native/wiki/Building-from-source) as a workaround.
> 
> 
> 
> Then you can also verify if this PR fixes the issue 🤓 

@Ragnar-H have you managed to build from source? Because I had to jump through hoops to test my piece of code.
Never managed to fully compile RN for android. And the official instructions are crap.

**Comment by fabriziobertoglio1987** [comment]
Instructions worked for me. You have to install ndk. I used Android studio and pointed env variable to my android_env/ndv path. On ubuntu

**Comment by danielgindi** [comment]
@fabriziobertoglio1987 I have the ndk installed and have set the ENV, but then all kinds of other errors started showing up. I'm on a mac. I tried this multiple times, never succeeded. Which lead me to patching an existing build just to test those tiny code changes.

**Comment by fabriziobertoglio1987** [comment]
@danielgindi [building from source](https://github.com/facebook/react-native/wiki/Building-from-source#building)

>If you don't want to install the required toolchain for building from source, you can use a prebuilt docker image to create a react native binary;
```bash
docker run --rm --name rn-build -v $PWD:/pwd -w /pwd reactnativecommunity/react-native-android /bin/sh -c "./gradlew installArchives"
```

**Comment by danielgindi** [comment]
> @danielgindi [building from source](https://github.com/facebook/react-native/wiki/Building-from-source#building)
> 
> 
> 
> >If you don't want to install the required toolchain for building from source, you can use a prebuilt docker image to create a react native binary;
> 
> ```bash
> 
> docker run --rm --name rn-build -v $PWD:/pwd -w /pwd reactnativecommunity/react-native-android /bin/sh -c "./gradlew installArchives"
> 
> ```
> 
> 

Everything required is installed. I actually prefer not to use docker when I need a working env. For CI that’s fine, but not for me :)
When (if?) I get some free time I’ll try again, and maybe open an issue about the errors.
I just gave up on it last time, also since every other commit also fails on CI so it’s not stable to begin with.

**Comment by danielgindi** [comment]
Okay so good news, bad news and good news:
1. Good news - I got this to compile locally
2. Bad news - this does not work. As RN seems to not call `super.onTouchEvent`, but pass the events at a later stage to the JS thread.
3. Now I don't have to due a POC on a separate project, but can work directly on RN so I'm working on a complete solution.

**Comment by ltcaosj** [comment]
@danielgindi that's a very good news though.

**Comment by danielgindi** [comment]
There's a failed test claiming that the package `com.facebook.react.views.view` does not exist, when that's clearly not true. If anyone knows what the heck, please take a look.

@kelset @mdvacca Could you guys please review this PR? It's now working perfectly for me, and it's an even easier change to digest and review. Much less code. And I'm still getting spammed by users asking me when this is coming to `master`, like I have any control over it.

**Comment by kelset** [comment]
>  like I have any control over it.

I'm in your same position 😓

**Comment by danielgindi** [comment]
@cpojer @shergin @hramos you guys are marked as reviewers for this, please have a look... :-)

**Comment by ltcaosj** [comment]
Looking forward to the release soon :D

**Comment by Rodrigo816** [comment]
omg I can't wait for this, thats amazing news.

**Comment by tolgacesur** [comment]
please have a look :) @cpojer @shergin @hramos

**Comment by danielgindi** [comment]
I'm sorry to say this, but obviously FB are only handling bugs and PRs that they have interest to. Which means things that are roadblocks in their on projects. Let's hope that they'll have this issue soon so they'll take a look.

**Comment by hsource** [comment]
This fix mostly worked, but I ran into an issue where it didn't work for a specific case. I have a newer patch in #29039 that I think is more complete:

While this fixed the issue on elements within the bounds of a parent with `overflow: visible`, it didn't fix the issue in a specific case I found where there were 3 nested views:

image

In the `findTouchTargetView` for View 1, it would see the touch is outside of View 2 and not traverse deeper. The fix in #29039 avoids that pitfall by continuing to traverse not only when `overflow` is visible, but in all cases. It then only does the bounds check later.

**Comment by danielgindi** [comment]
> This fix mostly worked, but I ran into an issue where it didn't work for a specific case. I have a newer patch in #29039 that I think is more complete:
> 
> 
> 
> While this fixed the issue on elements within the bounds of a parent with `overflow: visible`, it didn't fix the issue in a specific case I found where there were 3 nested views:
> 
> 
> 
> image
> 
> 
> 
> In the `findTouchTargetView` for View 1, it would see the touch is outside of View 2 and not traverse deeper. The fix in #29039 avoids that pitfall by continuing to traverse not only when `overflow` is visible, but in all cases. It then only does the bounds check later.

There is an option to suggest a change to this PR. Instead of creating a PR for each change anyone thinks a PR should have.

**Comment by hsource** [comment]
> > This fix mostly worked, but I ran into an issue where it didn't work for a specific case. I have a newer patch in #29039 that I think is more complete:
> 
> There is an option to suggest a change to this PR. Instead of creating a PR for each change anyone thinks a PR should have.

I originally started out with this Pull Request, but ended up diverging enough that I ended up just making the changes on the originals. Thanks a lot for this pull request though! I referenced its logic a lot to figure out what I should start with.

**Comment by danielgindi** [comment]
@hsource I've just tested, and this PR works for me with nested views too.

**Comment by hsource** [comment]
> @hsource I've just tested, and this PR works for me with nested views too.

I just added a test case for the specific case into my Pull Request #29039 - the box "D" isn't tappable with this change, but should be.
 
Specific test case

**Comment by danielgindi** [comment]
> > @hsource I've just tested, and this PR works for me with nested views too.
> 
> 
> 
> I just added a test case for the specific case into my Pull Request #29039 - the box "D" isn't tappable with this change, but should be.
> 
>  
> 
> Specific test case

Thanks.
I’ll take a look.

**Comment by lgenzelis** [comment]
Any updates? This is **really** important for android. :(

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java [hunk 1]
```diff
@@ -115,31 +114,38 @@ private static View findTouchTargetView(float[] eventCoords, ViewGroup viewGroup
...
     for (int i = childrenCount - 1; i >= 0; i--) {
       int childIndex =
           zIndexedViewGroup != null ? zIndexedViewGroup.getZIndexMappedChildIndex(i) : i;
       View child = viewGroup.getChildAt(childIndex);
-      PointF childPoint = mTempPoint;
+      TouchTestResult touchTestData = mTouchTestData;
+
       if (isTransformedTouchPointInView(
-          eventCoords[0], eventCoords[1], viewGroup, child, childPoint)) {
+          eventCoords[0], eventCoords[1], viewGroup, child, touchTestData, allowOverflow)) {
```
Comment (fabriziobertoglio1987) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:130]:
```suggestion
          eventCoords[0], eventCoords[1], viewGroup, child, allowOverflow)) {
```

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java [hunk 2]
```diff
@@ -148,7 +154,9 @@ private static View findTouchTargetView(float[] eventCoords, ViewGroup viewGroup
    * ViewGroup#isTransformedTouchPointInView()}
    */
   private static boolean isTransformedTouchPointInView(
-      float x, float y, ViewGroup parent, View child, PointF outLocalPoint) {
+          float x, float y, ViewGroup parent, View child,
+          TouchTestResult outTestData,
```
Comment (fabriziobertoglio1987) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:158]:
```suggestion
```

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java [hunk 3]
```diff
@@ -162,25 +170,28 @@ private static boolean isTransformedTouchPointInView(
...
-    } else {
-      if ((localX >= 0 && localX < (child.getRight() - child.getLeft()))
-          && (localY >= 0 && localY < (child.getBottom() - child.getTop()))) {
-        outLocalPoint.set(localX, localY);
-        return true;
-      }
+    boolean hittest = isLocalCoordinateInView(localX, localY, child);
+
+    if (allowOverflow || hittest) {
+      outTestData.set(localX, localY, hittest);
```
Comment (fabriziobertoglio1987) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:177]:
```suggestion
      mTouchTestData.set(localX, localY, hittest);
```

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java [hunk 4]
```diff
@@ -115,31 +114,38 @@ private static View findTouchTargetView(float[] eventCoords, ViewGroup viewGroup
...
         // If it is contained within the child View, the childPoint value will contain the view
         // coordinates relative to the child
         // We need to store the existing X,Y for the viewGroup away as it is possible this child
         // will not actually be the target and so we restore them if not
         float restoreX = eventCoords[0];
         float restoreY = eventCoords[1];
-        eventCoords[0] = childPoint.x;
-        eventCoords[1] = childPoint.y;
-        View targetView = findTouchTargetViewWithPointerEvents(eventCoords, child);
+        eventCoords[0] = touchTestData.x;
```
Comment (fabriziobertoglio1987) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:137]:
```suggestion
        eventCoords[0] = mTouchTestData.x;
```

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java [hunk 5]
```diff
@@ -115,31 +114,38 @@ private static View findTouchTargetView(float[] eventCoords, ViewGroup viewGroup
...
+        eventCoords[1] = touchTestData.y;
```
Comment (fabriziobertoglio1987) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:138]:
```suggestion
        eventCoords[1] = mTouchTestData.y;
```

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java [hunk 6]
```diff
@@ -115,31 +114,38 @@ private static View findTouchTargetView(float[] eventCoords, ViewGroup viewGroup
...
+        View targetView = findTouchTargetViewWithPointerEvents(eventCoords, child, touchTestData.hittest);
```
Comment (fabriziobertoglio1987) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:139]:
```suggestion
        View targetView = findTouchTargetViewWithPointerEvents(eventCoords, child, mTouchTestData.hittest);
```

Comment (danielgindi) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:139]: Why do you care if it has a local variable to point to the actual static one?
Did you generate release mode code and investigate it to see if it actually costs something?

Comment (fabriziobertoglio1987) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:139]: @danielgindi never mind

Comment (fabriziobertoglio1987) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:139]: @danielgindi I am learning this now as I'm a Ruby Developer. 

>Why do you care if it has a local variable to point to the actual static one?

As you can read [here](https://stackoverflow.com/a/70358/11452286) static nested class, means the class is saved as a static property of the parent, in your case would be `TouchTargetHelper.TouchTestResult`

>Static nested classes are accessed using the enclosing class name:

```java
TouchTargetHelper.TouchTestResult
```
>For example, to create an object for the static nested class, use this syntax: (from outside that class)

```java
TouchTargetHelper.TouchTestResult nestedObject = new TouchTargetHelper.TouchTestResult();
```
My comment has nothing to do with the above and I believe you misunderstood my point.

I am not referring to the `static class TouchTestResult`, but how you are using the local and instance variables of the TouchTargetHelper class. 

>Did you generate release mode code and investigate it to see if it actually costs something?

With my regards
I wish you a good day
Fabrizio Bertoglio

**Comment by danielgindi** [comment]
@hsource The fix to an issue with box-only was actually quite simple, moving 3 lines of code somewhere else. It was considering the wrong overflow property. I borrowed your test to demonstrate.

**Comment by shergin** [comment]
Yeah, this is very important issue to fix.

Can we start from decoupling the example from this diff for quick landing? We think we solved this issue in Fabric on iOS and Android, and the test will be very helpful to verify that.

Then, I think @mdvacca has the most context to review the PR. :)

**Comment by hsource** [comment]
> Can we start from decoupling the example from this diff for quick landing? We think we solved this issue in Fabric on iOS and Android, and the test will be very helpful to verify that.

That sounds great! I didn't realize Fabric was so close.

I'm the original author of both another fix (#29039) and the examples used in this change. I just created a PR with just the examples - take a look at #29393

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactClippingViewGroup.java [hunk 7]
```diff
@@ -54,4 +56,13 @@
 
   /** Get the current value of {@code removeClippedSubviews} property. */
   boolean getRemoveClippedSubviews();
+
+  /**
+   * Sets property {@code overflow} as a result of style update in JS.
+   */
+  void setOverflow(String overflow);
```
[the remark this answers is not in the document]
Comment (danielgindi) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactClippingViewGroup.java:63]: the `get` and `set` are independent, that's always the case, but they are related. And in other similar react interfaces the getters/setters come together. I'm trying to adhere to the style around it.

I intentionally added them to `ReactClippingViewGroup` as overflow is directly related to clipping.

Separating interface for everything that *could* be technically separated - that's not am effective approach, especially for Android. It comes with a cost in a Java environment.

File: RNTester/js/examples/PointerEvents/PointerEventsExample.js [hunk 8]
```diff
@@ -191,6 +254,12 @@ const exampleClasses: Array<ExampleClass> = [
     description:
       "`box-only` causes touch events on the container's child components to pass through and will only detect touch events on the container itself.",
   },
+  {
+    Component: OverflowVisibleExample,
+    title: '`overflow: visible`',
+    description:
+      '`overflow: visible` style should allow subelements that are outside of the parent box to be touchable. Tapping the parts of Box B outside Box A should print "B touched" and "A touched", and tapping Box C should also print "C touched" and "A touched".',
+  },
```
[the remark this answers is not in the document]
Comment (danielgindi) [reply RNTester/js/examples/PointerEvents/PointerEventsExample.js:262]: Yeah I've added it only when testing locally, then decided to add it to the repo and forgot to add back the component to the list.

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java (hunk 1 above)
Comment (danielgindi) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java]: There was a tiny bug in the overflow test.
If you have change suggestions, please re-review them on the updated version.

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java (hunk 3 above)
Comment (danielgindi) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:175]: There was a tiny bug in the overflow test.
If you have change suggestions, please re-review them on the updated version.

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java (hunk 4 above)
Comment (danielgindi) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:133]: There was a tiny bug in the overflow test.
If you have change suggestions, please re-review them on the updated version.

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java (hunk 5 above)
Comment (danielgindi) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java:134]: There was a tiny bug in the overflow test.
If you have change suggestions, please re-review them on the updated version.

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java (hunk 2 above)
Comment (danielgindi) [reply ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java]: There was a tiny bug in the overflow test.
If you have change suggestions, please re-review them on the updated version.

**Comment by danielgindi** [comment]
> In #29025, I turn on and off Gravity.LEFT based on the value of mGravityReset in the onFocusChanged callback. If you need to cache mTouchTestData to use it later in one of the Android callbacks, may make sense to use an instance variable..

The use of the `static TouchTestResult mTouchTestData` variable is not about caching it for later use. It's about not reallocating the objet hundreds of times for each tap. There's just no point in it.
The signature of the `isTransformedTouchPointInView` function is such so that it's more generic and can be reused in other ways later. This makes it clear who uses or changes a value, and who owns or manages the static `mTouchTestData` value.

**Comment by stackia** [comment]
Any updates? This make it really difficult to implement 'tooltip' effect

**Comment by danielgindi** [comment]
@stackia don't count on it. But in the mean time you could add a transparent overlay on top of everything, and move tooltips around inside it. 
Your case is pretty simple as tooltips usually have to be the highest z-index, so you don't *really* need overflowing.

**Comment by stackia** [comment]
> @stackia don't count on it. But in the mean time you could add a transparent overlay on top of everything, and move tooltips around inside it.
> Your case is pretty simple as tooltips usually have to be the highest z-index, so you don't _really_ need overflowing.

It's difficult to implement a tooltip component that syncs with FlatList scrolling without using overflow.

**Comment by McSam94** [comment]
any update please? 😔

**Comment by aubert-creation** [comment]
can we hope to have this fix in a future release ?

**Comment by danielpoliveira** [comment]
any news about the bug fix? 😞

**Comment by Drnutsu** [comment]
This one is so important issue, please review.

**Comment by Svarto** [comment]
Same problem, would be great with a fix!

**Comment by summerkiflain** [comment]
This PR is super important for implementing absolute positioned elements in layout, I hope someone merges it soon.

**Comment by lgenzelis** [comment]
Any updates about this? It would be really helpful to have this merged :)

**Comment by FrancisLeigh** [comment]
Branch has conflicts... are we able to get a merge!?

**Comment by danielgindi** [comment]
> Branch has conflicts... are we able to get a merge!?

Of course, after months of no response from Facebook, there should be conflicts...
I'll rebase soon when I get a chance

**Comment by tasosxak** [comment]
Very important feature.
Thank you.

**Comment by FrancisLeigh** [comment]
Stil hyped for this feature!

**Comment by afterburn** [comment]
bump

**Comment by pirroman** [comment]
It's very important fix! Hope to see this PR merged soon.

**Comment by danilo900** [comment]
Is that already merged?

**Comment by danilo900** [comment]
I have tried it out and it still not fixing.

I have a view outside the viewport, so, I translate it position to the middle of the viewport, but nothing is triggering the click event

**Comment by danielgindi** [comment]
> I have tried it out and it still not fixing.
> 
> 
> 
> I have a view outside the viewport, so, I translate it position to the middle of the viewport, but nothing is triggering the click event

If you have a situation where it does not work, please reproduce it in the tests. There are already tests for views outside other views, with various nested levels.

**Comment by alfalcon90** [comment]
Any updates on this? Any way that we can fix this without waiting on FB?

**Comment by jstheoriginal** [comment]
> Any updates on this? Any way that we can fix this without waiting on FB?

It's always possible to build react native from source, pointing to the patched branch. 

Instructions: https://github.com/facebook/react-native/wiki/Building-from-source

**Comment by ashmanouski** [comment]
why it's not merged yet? more than 2 years for this bug and no response from FB

**Comment by ltcaosj** [comment]
I think lots of people is waiting for this fix.

**Comment by danielgindi** [comment]
> I think lots of people is waiting for this fix.

If someone from Facebook will say that they are interested in merging this - I'd be willing to rebase on `master`. Until then - I have no further interest in this. I have lost faith in Facebook's dev team.

**Comment by a-eid** [comment]
I'm wondering how could I test this PR.

**Comment by aubert-creation** [comment]
> > I think lots of people is waiting for this fix.
> 
> If someone from Facebook will say that they are interested in merging this - I'd be willing to rebase on `master`. Until then - I have no further interest in this. I have lost faith in Facebook's dev team.

Maybe with the rise of Flutter, Facebook will start to move its buttocks

**Comment by Nico-Guyon** [comment]
I wonder how much time we all lost with this android unability to send Touch events to overflowed objects.

It's quite sad to see that this still hasn't been fixed, since this PR from Sept 2019 :/

**Comment by ltcaosj** [comment]
Same to me, it's sad.

**Comment by ArshanKhanifar** [comment]
+1, why isn't this getting fixed? 😂

**Comment by danielgindi** [comment]
Finally, the PR by @hsource was merged. This older one is now obsolete.
My merged branch will still be available for use until this will be released, then I'll delete it.

**Comment by himanshubishtonecode** [comment]
> Hi, maybe somebody know when this PR will be merged ?

Today it will for sure.

**Pull request closed without merging**